pkg/pipeline/decode/decode_json.go

```diff
@@ -42,6 +43,9 @@ func (c *DecodeJson) Decode(in []interface{}) []config.GenericMap {
 			continue
 		}
 		decodedLine2 := make(config.GenericMap, len(decodedLine))
+		// flows directly ingested by flp-transformer won't have this field, so we need to add it
+		// here. If the received line already contains the field, it will be overridden later
+		decodedLine2["TimeReceived"] = time.Now().Unix()
```

Comment:
May be we should add a check to ensure there is no TimeReceived field?

Reply:
In that case, if there is no `TimeReceived` field, it will be added to the current payload. If there is an actual `TimeReceived` field, it will be added after this, overriding the value that we set here.

Reply:
Is it what we want? To override TimeReceived if the field is already here?